sdk-lib/src/main/java/cash/z/ecc/android/sdk/type/WalletTypes.kt

```diff
@@ -52,16 +52,16 @@ data class WalletBirthday(
 }
 
 /**
- * A grouping of keys that correspond to a single wallet account but do not have spend authority.
+ * A [ZIP 316] Unified Full Viewing Key, corresponding to a single wallet account.
  *
- * @param extfvk the extended full viewing key which provides the ability to see inbound and
- * outbound shielded transactions. It can also be used to derive a z-addr.
- * @param extpub the extended public key which provides the ability to see transparent
- * transactions. It can also be used to derive a t-addr.
+ * A `UnifiedFullViewingKey` has the authority to view transactions for an account, but
+ * does not have spend authority. It can be used to derive a [UnifiedAddress] for the
+ * account.
+ *
+ * @param[encoding] The string encoding of the UFVK.
  */
-data class UnifiedViewingKey(
-    val extfvk: String = "",
-    val extpub: String = ""
+data class UnifiedFullViewingKey(
+    val encoding: String = ""
```

Comment:
This is a pre-existing issue.  I don’t think we ever want to allow `UnifiedFullViewingKey(“”)` 

Since we’re already making a breaking change here, can we drop the empty string as a default constructor parameter?

Reply:
Also are there any invariants that you’d like to enforce here?  (Minimum length?)

Reply:
I notice a security issue too.  We should probably drop `data` so this becomes an ordinary class. Data provides a default equals, hash code, and toString implementation.  The `toString` is dangerous in terms of accidentally logging sensitive information. (Alternative is to override the toString to block its default return value).

Reply:
This is actually something I want to overhaul before we make the breaking release. I agree with making this an opaque class, so we can better control how it is instantiated, and alter its internals freely without breaking changes. I'm not sure I want to do that in this PR though, which is focused on the UFVK breaking API changes that need to affect the FFI. I'll open an issue for addressing this.

Reply:
#531

Reply:
As we're now using a feature branch, we can ensure that we complete #531 before merging it.